routers/web/repo/view.go

```diff
@@ -728,16 +728,65 @@ func checkHomeCodeViewable(ctx *context.Context) {
 	ctx.NotFound("Home", fmt.Errorf(ctx.Tr("units.error.no_unit_allowed_repo")))
 }
 
+func checkCitationFile(ctx *context.Context) {
+	// Check if repo is not empty and if we are on repo src
+	if ctx.Repo.Repository.IsEmpty {
+		return
+	}
+	entry, err := ctx.Repo.Commit.GetTreeEntryByPath(ctx.Repo.TreePath)
```

Comment:
we do this at `renderCode` too ...

so now we have have duplicated action for a single page view :/